clang/lib/Driver/ToolChains/Arch/AArch64.cpp

```diff
@@ -321,9 +321,11 @@ void aarch64::getAArch64TargetFeatures(const Driver &D,
     }
   }
 
-  if (Arg *A = Args.getLastArg(options::OPT_mno_unaligned_access,
-                               options::OPT_munaligned_access)) {
-    if (A->getOption().matches(options::OPT_mno_unaligned_access))
+  if (Arg *A = Args.getLastArg(
+          options::OPT_mstrict_align, options::OPT_mno_strict_align,
+          options::OPT_mno_unaligned_access, options::OPT_munaligned_access)) {
+    if (A->getOption().matches(options::OPT_mstrict_align) ||
+        A->getOption().matches(options::OPT_mno_unaligned_access))
```

Comment:
Why keep `mno_unaligned_access` for AArch64 while remove it from RISC-V and LoongArch?

In fact `mno_unaligned_access` is not supported by GCC for  AArch64?

Reply:
For MIPSr6, since `mno_unaligned_access` has been in GCC for almost 2 years, I prefer support these 2 options both for LLVM and GCC.

In fact, I don't think that it is a bad idea to support these 2 options both: it will make life easier for users.

Reply:
If we prefer strict-align, we may claim `mno_unaligned_access` as obsolete in our documents.

Reply:
AArch64 supports Apple OSes. I am unclear whether have any unintended `-munaligned-access`/`-mno-unaligned-access`. I'd like to remove them as well, but probably later.

For RISC-V and LoongArch, they have Linux support and the GCC influence is very strong. The Clang support is also relatively new. Removing the alias likely cause no disruption at all.

Reply:
LGTM.

I will add `strict_align` for MIPSr6 to GCC, and rebase my PR to LLVM.

Reply:
Sounds good. When you implement -munaligned-access for LLVM's MIPS port, be sure to update the HelpText in `clang/include/clang/Driver/Options.td` `AArch32/MIPS only`

Reply:
Preventing unaligned access can be useful in AArch64, it is an option we do use to build our embedded C-libraries with (not a focus for GCC). It is documented in the toolchain manual https://developer.arm.com/documentation/101754/0621/armclang-Reference/armclang-Command-line-Options/-munaligned-access---mno-unaligned-access

In summary, we'd like to keep it for AArch64.

AArch64 always has the option of using unaligned accesses, but they can be disabled by writing the SCTLR register, and accesses to Device memory always need to be aligned. Code that runs before the MMU is enabled runs as if Device memory.

```
Unaligned accesses to Normal memory
The behavior of unaligned accesses to Normal memory is dependent on all of the following:
• The instruction causing the memory access.
• The memory attributes of the accessed memory.
• The value of SCTLR_ELx.{A, nAA}.
• Whether or not FEAT_LSE2 is implemented.
```

Reply:
Thanks for the AArch64 notes. 
This HelpText (`AArch32 only`) change is to discourage `-munaligned-access` for AArch64, since GCC doesn't support `-munaligned-access`.

Personally I assume that most modern architectures should switch to `-mstrict-align`.

(Related, I have been trying to make more code compliant under `-fsanitize=alignment`, which might makes memory access operations with hardware check easier. 
)